imx943: pm_mcore.c causes link failure when NXP SCMI CPU domain helpers are not enabled (undefined scmi_nxp_cpu_* references)

## Describe the bug
Building for i.MX943 M33 can fail at link time with undefined references from `soc/nxp/imx/imx9/imx943/pm_mcore.c` to NXP SCMI CPU-domain helper APIs.

Example failing symbols (from `pm_state_before()/pm_state_set()/pm_state_exit_post_ops()`):
- `scmi_nxp_cpu_pd_lpm_set`
- `scmi_nxp_cpu_set_irq_mask`
- `scmi_nxp_cpu_sleep_mode_set`

The helpers are implemented in `drivers/firmware/scmi/nxp/cpu.c`, but that file is only built when `CONFIG_NXP_SCMI_CPU_DOMAIN_HELPERS=y` (which itself depends on `CONFIG_ARM_SCMI_NXP_VENDOR_EXTENSIONS` and `DT_HAS_NXP_SCMI_CPU_ENABLED`).

However `pm_mcore.c` is compiled when `CONFIG_SOC_IMX943_PM_MCORE=y` (default y when `PM` is enabled for i.MX943 M33/M7), even if the SCMI CPU helpers are not enabled. This leads to the undefined reference link failures.

Reference line in upstream:
https://github.com/zephyrproject-rtos/zephyr/blob/0bb9a469c1dbe1dfb825cd68a08fab8b70c3701e/soc/nxp/imx/imx9/imx943/pm_mcore.c#L139

## To Reproduce
Observed in CI/twister build:
- Test: `cpp.main.newlib.cxx_tests.new_delete`
- Zephyr version: `v4.3.0-5931-g8100484e5d43`
- Platform: `nxp_virtual/mimx94398/m33`

Linker error excerpt:
```
ld.bfd: zephyr/libzephyr.a(pm_mcore.c.obj): in function `pm_state_before':
.../soc/nxp/imx/imx9/imx943/pm_mcore.c:208: undefined reference to `scmi_nxp_cpu_pd_lpm_set'
.../pm_mcore.c:229: undefined reference to `scmi_nxp_cpu_set_irq_mask'
.../pm_mcore.c:280: undefined reference to `scmi_nxp_cpu_sleep_mode_set'
.../pm_mcore.c:304: undefined reference to `scmi_nxp_cpu_sleep_mode_set'
collect2: error: ld returned 1 exit status
```

## Expected behavior
Either:
1. `pm_mcore.c` is not built unless the required SCMI CPU helpers are enabled, **or**
2. `pm_mcore.c` is guarded so it does not reference these symbols when the helpers are not present.

## Proposed fix
In `soc/nxp/imx/imx9/imx943/Kconfig`, make `SOC_IMX943_PM_MCORE` depend on `NXP_SCMI_CPU_DOMAIN_HELPERS`.

Example:
```Kconfig
config SOC_IMX943_PM_MCORE
	bool "i.MX943 multi-core PM shared logic"
	default y
	depends on PM
	depends on SOC_MIMX94398_M33 || SOC_MIMX94398_M7_0 || SOC_MIMX94398_M7_1
	depends on NXP_SCMI_CPU_DOMAIN_HELPERS
```

This ensures the SoC PM implementation is only built when the implementation of `scmi_nxp_cpu_*` helpers is available.

## Additional context
- Definitions are in: `drivers/firmware/scmi/nxp/cpu.c`
- Built under: `CONFIG_NXP_SCMI_CPU_DOMAIN_HELPERS` (`drivers/firmware/scmi/nxp/CMakeLists.txt`)

## Environment
- Toolchain: zephyr-sdk 0.17.1-rc4 (arm-zephyr-eabi gcc 12.2.0)
- Host: Linux (Jenkins CI)
- version: https://github.com/zephyrproject-rtos/zephyr/commit/2fd6d9ad8d36918d98fd2747a9b4fb8f3c07c971

--- a/soc/nxp/imx/imx9/imx943/Kconfig
+++ b/soc/nxp/imx/imx9/imx943/Kconfig
@@ -58,5 +58,6 @@ config SOC_IMX943_PM_MCORE
 	default y
 	depends on PM
 	depends on SOC_MIMX94398_M33 || SOC_MIMX94398_M7_0 || SOC_MIMX94398_M7_1
+	depends on NXP_SCMI_CPU_DOMAIN_HELPERS
 	help
 	  Common PM flow for Cortex-M cores (M33/M7) on i.MX943

